### ISSUE TITLE
locksmith.rb:327: NoMethodError:  undefined method `+' for nil:NilClass

### ISSUE THREAD

**Issue by rgaufman** [mhenrixon/sidekiq-unique-jobs#686]
**Describe the bug**
I am getting the following exception.

**Expected behavior**
It should not throw an exception

**Current behavior**
Throws an exception

```
2022-03-02 11:43:39.410252 W [14198:processor] Sidekiq -- NoMethodError: undefined method `+' for nil:NilClass
2022-03-02 11:43:39.410298 W [14198:processor] Sidekiq -- /data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:327:in `drift'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:331:in `add_drift'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:244:in `primed_async'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:196:in `call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:196:in `block in lock!'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:228:in `enqueue'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:193:in `lock!'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:105:in `block in execute'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq.rb:100:in `block in redis'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.2.5/lib/connection_pool.rb:63:in `block (2 levels) in with'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.2.5/lib/connection_pool.rb:62:in `handle_interrupt'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.2.5/lib/connection_pool.rb:62:in `block in with'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.2.5/lib/connection_pool.rb:59:in `handle_interrupt'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.2.5/lib/connection_pool.rb:59:in `with'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq.rb:97:in `redis'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/connection.rb:19:in `redis'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/locksmith.rb:104:in `execute'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/lock/until_executed.rb:36:in `execute'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/middleware/server.rb:25:in `call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/middleware.rb:41:in `block in call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/logging.rb:132:in `block in with_logging_context'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/logger.rb:11:in `with'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/logging.rb:147:in `call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/logging.rb:147:in `with_configured_loggers_context'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/logging.rb:131:in `with_logging_context'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-unique-jobs-7.1.15/lib/sidekiq_unique_jobs/middleware.rb:40:in `call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/middleware/chain.rb:140:in `block in invoke'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/airbrake-ruby-6.0.1/lib/airbrake-ruby/benchmark.rb:13:in `measure'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/airbrake-12.0.0/lib/airbrake/sidekiq.rb:10:in `call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/middleware/chain.rb:140:in `block in invoke'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/middleware/chain.rb:143:in `invoke'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:163:in `block in process'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:136:in `block (6 levels) in dispatch'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/job_retry.rb:114:in `local'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:135:in `block (5 levels) in dispatch'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/rails.rb:14:in `block in call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/execution_wrapper.rb:88:in `wrap'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/reloader.rb:72:in `block in wrap'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/execution_wrapper.rb:88:in `wrap'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/reloader.rb:71:in `wrap'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/rails.rb:13:in `call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:131:in `block (4 levels) in dispatch'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:257:in `stats'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:126:in `block (3 levels) in dispatch'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/job_logger.rb:13:in `call'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:125:in `block (2 levels) in dispatch'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/job_retry.rb:81:in `global'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:124:in `block in dispatch'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/job_logger.rb:39:in `prepare'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:123:in `dispatch'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:162:in `process'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:78:in `process_one'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/processor.rb:68:in `run'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/util.rb:56:in `watchdog'
/data/deployer/timeagent/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.4.1/lib/sidekiq/util.rb:65:in `block in safe_thread'
```

**Worker class**
All

**Additional context**
These are the parameters that AirBrake captured:

```
{
  "job": {
    "args": [
      {
        "action": "create",
        "camera_id": "CAMERAID",
        "controller": "api/v3/snapshots",
        "filename": "CAMERAID/20220302_10/20220302_102704-0.000-0.jpg",
        "in_schedule": true,
        "motion_percent": "0",
        "object_number": "0",
        "objects": [],
        "severity": "info",
        "snapshot": {
          "camera_id": "CAMERAID",
          "filename": "CAMERAID/20220302_10/20220302_102704-0.000-0.jpg",
          "in_schedule": true,
          "motion_percent": "0",
          "object_number": "0",
          "objects": [],
          "severity": "info",
          "state": "snapshot",
          "thumbnail": "CAMERAID/20220302_10/thumbs/20220302_102704-0.000-0.thumb.jpg",
          "timestamp": "1646216824",
          "token": "TOKEN"
        },
        "state": "snapshot",
        "thumbnail": "CAMERAID/20220302_10/thumbs/20220302_102704-0.000-0.thumb.jpg",
        "timestamp": "1646216824",
        "token": "TOKEN"
      }
    ],
    "backtrace": true,
    "class": "SnapshotCreateWorker",
    "created_at": 1646216836.039246,
    "enqueued_at": 1646216836.044716,
    "jid": "6ee21d26e68697968de4820e",
    "lock": "until_executed",
    "lock_args": [
      {
        "action": "create",
        "camera_id": "CAMERAID",
        "controller": "api/v3/snapshots",
        "filename": "CAMERAID/20220302_10/20220302_102704-0.000-0.jpg",
        "in_schedule": true,
        "motion_percent": "0",
        "object_number": "0",
        "objects": [],
        "severity": "info",
        "snapshot": {
          "camera_id": "CAMERAID",
          "filename": "CAMERAID/20220302_10/20220302_102704-0.000-0.jpg",
          "in_schedule": true,
          "motion_percent": "0",
          "object_number": "0",
          "objects": [],
          "severity": "info",
          "state": "snapshot",
          "thumbnail": "CAMERAID/20220302_10/thumbs/20220302_102704-0.000-0.thumb.jpg",
          "timestamp": "1646216824",
          "token": "TOKEN"
        },
        "state": "snapshot",
        "thumbnail": "CAMERAID/20220302_10/thumbs/20220302_102704-0.000-0.thumb.jpg",
        "timestamp": "1646216824",
        "token": "TOKEN"
      }
    ],
    "lock_digest": "uniquejobs:4506eb00cd9246dd4534a4e3d79ad52f",
    "lock_prefix": "uniquejobs",
    "lock_timeout": null,
    "lock_ttl": 3600,
    "queue": "default",
    "retry": false,
    "unique": "until_and_while_executing",
    "unique_args": "#<Proc:0x00007f94e87bcaa8 /data/deployer/timeagent/config/initializers/tasks.rb:12 (lambda)>"
  }
}
```

Any ideas please?

**Comment by mhenrixon** [comment]
Should be fixed by #682

**Comment by arekk** [comment]
+1 for this
